lib/rails_state_machine/model.rb

```diff
@@ -40,6 +40,10 @@ def state_machine_state_manager(state_attribute)
       @state_machine_state_managers[state_attribute] ||= StateManager.new(self, state_machine(state_attribute), state_attribute)
     end
 
+    def clear_state_machine_state_managers_cache
+      @state_machine_state_managers = nil
+    end
+
```

Comment:
Can you get rid of `@state_machine_state_managers ||= {}`, if `clear_state_machine_state_managers_cache` initializes it with an empty hash? I'd like to keep the code paths for "a new state machine" and "a duplicated state machines" identical.

```suggestion
    def state_machine_state_manager(state_attribute)
      @state_machine_state_managers[state_attribute] ||= StateManager.new(self, state_machine(state_attribute), state_attribute)
    end

    def clear_state_machine_state_managers_cache
      @state_machine_state_managers = {}
    end

```